microsoftgraph/msgraph-cli-archived
# Feat/support beta extensions

## Overview

Supports users switching between **beta** and **v1.0**

### Demo

**Selecting Graph Version**
![image](https://user-images.githubusercontent.com/12011447/100873488-59c87880-34b4-11eb-848d-51de10261551.png)

**Show current graph version**
![image](https://user-images.githubusercontent.com/12011447/100873660-8da39e00-34b4-11eb-919e-76b8bec35928.png)

### Notes
The CLI will ship with both v1.0 and beta extensions then we use the graph version property to switch between beta and v1.0 extensions. The CLI loads v1.0 extensions by default.

## Testing Instructions

* Pull these changes
* Build and run the docker container, instructions are in the readme.
* Run `mg graph-version select` to select the graph version
* Run `mg graph-version show-current` to see the current graph version

## Files to review
* build_scripts/generate_extensions.py
* build_scripts/install_extensions.py
* msgraph/command_modules/profile/custom.py
* msgraph/cli/core/commands/client_factory.py

Closes #53
